### ISSUE TITLE
Unity Object null handling severity should be warning by default.

### ISSUE THREAD

**Issue by sator-imaging** [microsoft/Microsoft.Unity.Analyzers#217]
#### Problem statement

Unity Object null handling (UnityObjectNullHandling.cs) severity is "Info" by default. It says should not use.
I know it's not critical because the following code works as expected.

```csharp
UnityEngine.GameObject unityObjectIsNull = null;
Debug.Log(unityObjectIsNull?.name);
```

But, in production, null handling code like as above *sometimes* causes Null reference error.
It's nice to change the severity is warning by default.

#### Proposed solution

UNT0007, UNT0008, UNT0023 severity is set to warning by default.

**Comment by sailro** [comment]
Hello, this is something you can completely customize using an `.editorconfig` file:

```ini
[*.cs]
dotnet_diagnostic.UNT0007.severity = warning
dotnet_diagnostic.UNT0008.severity = warning
dotnet_diagnostic.UNT0023.severity = warning
```
